clouddrift/adapters/gdp/gdpsource.py

```diff
@@ -0,0 +1,544 @@
+    "posObsDay": np.float64,
+    "posObsYear": np.float32,
+    "latitude": np.float32,
+    "longitude": np.float32,
+    "qualityIndex": np.float32,
+    "senObsMonth": np.float32,
+    "senObsDay": np.float64,
+    "senObsYear": np.float32,
+    "drogue": np.float32,
+    "sst": np.float32,
+    "voltage": np.float32,
+    "sensor4": np.float32,
+    "sensor5": np.float32,
+    "sensor6": np.float32,
+}
+
+_INPUT_COLS_PREFILTER_DTYPES = {
+    "posObsMonth": np.str_,
+    "posObsYear": np.float64,
+    "senObsMonth": np.str_,
+    "senObsYear": np.float64,
+    "drogue": np.str_,
+}
+
+
+VARS_ATTRS: dict = {
+    "id": {"long_name": "Global Drifter Program Buoy ID", "units": "-"},
+    "rowsize": {
+        "long_name": "Number of observations per trajectory",
+        "sample_dimension": "obs",
+        "units": "-",
+    },
+    "wmo_number": {
+        "long_name": "World Meteorological Organization buoy identification number",
+        "units": "-",
+    },
+    "program_number": {
+        "long_name": "Current Program",
+        "units": "-",
+    },
+    "buoys_type": {
+        "long_name": "Buoy type (see https://www.aoml.noaa.gov/phod/dac/dirall.html)",
+        "units": "-",
+    },
+    "start_date": {
+        "long_name": "First good date and time derived by DAC quality control",
+    },
+    "start_lon": {
+        "long_name": "First good longitude derived by DAC quality control",
+        "units": "degrees_east",
+    },
+    "start_lat": {
+        "long_name": "First good latitude derived by DAC quality control",
+        "units": "degrees_north",
+    },
+    "end_date": {
+        "long_name": "Last good date and time derived by DAC quality control",
+    },
+    "end_lon": {
+        "long_name": "Last good longitude derived by DAC quality control",
+        "units": "degrees_east",
+    },
+    "end_lat": {
+        "long_name": "Last good latitude derived by DAC quality control",
+        "units": "degrees_north",
+    },
+    "drogue_off_date": {
+        "long_name": "Date and time of drogue loss",
+    },
+    "death_code": {
+        "long_name": "Type of death",
+        "units": "-",
+        "comments": "0 (buoy still alive), 1 (buoy ran aground), 2 (picked up by vessel), 3 (stop transmitting), 4 (sporadic transmissions), 5 (bad batteries), 6 (inactive status)",
+    },
+    "position_datetime": {
+        "comments": "Position datetime derived from the year, month, day and time (represented as a ratio of a day) columns found in the source dataset that represent when the position of the drifter was measured. This value is only different from the sensor_datetime when the position of the drifter was determined onboard the Argos satellites using the doppler shift.",
+    },
+    "sensor_datetime": {
+        "comments": "Sensor datetime derived from the year, month, day and time (represented as a ratio of a day) columns found in the source dataset that represent when the sensor (like temp) data is recorded",
```

Comment:
Sensor datetime derived from the year, month, and day (represented as a ratio of a day) when sensor data were recorded.